addons/account_peppol/models/res_partner.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import requests
+from lxml import etree
+from hashlib import md5
+from urllib import parse
+
+from odoo import api, fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    account_peppol_is_endpoint_valid = fields.Boolean(
+        string="PEPPOL endpoint validity",
+        help="The partner's EAS code and PEPPOL endpoint are valid",
+        compute="_compute_account_peppol_is_endpoint_valid",
+        store=True,
+    )
+    account_peppol_validity_last_check = fields.Date(
+        string="Checked on",
+        help="Last Peppol endpoint verification",
+        readonly=True,
+    )
+
+    @api.depends('peppol_eas', 'peppol_endpoint')
+    def _compute_account_peppol_is_endpoint_valid(self):
+        for partner in self:
+            # Every change in peppol_eas or peppol_endpoint should set the validity back to False
+            partner.account_peppol_is_endpoint_valid = False
+
+    def _get_resolvable_smp(self, participant):
```

Comment:
don't forget to prefix all methods by peppol when working on a low-level generic model like partners